CMakeLists.txt

```diff
@@ -394,6 +394,10 @@ if(USE_NGHTTP2)
   find_package(NGHTTP2 REQUIRED)
   include_directories(${NGHTTP2_INCLUDE_DIRS})
   list(APPEND CURL_LIBS ${NGHTTP2_LIBRARIES})
+  if(NOT BUILD_SHARED_LIBS)
```

Comment:
Does't `BUILD_SHARED_LIBS` imply that you build libcurl statically and it doesn't actually say anything about the dependencies?

Reply:
Indeed, `BUILD_SHARED_LIBS` only affects `add_library` and doesn't affect `find_library` and other, but I could not think of a better variable to tie it to other than creating a new option, something like `STATIC_NGHTTP2` or similar to force static compilation by finding the static libs and cflags, but I am not sure if you guys are fine with adding new cmake options.

I would much prefer to add something like a `STATIC_${DEPS}` options, but would like to hear feedback about something like that, especially since it not something that is represented in the configure file.

Reply:
Something like https://github.com/1480c1/curl/commit/37ada3e08a808c98beec172b9da376aa0b3d9ca2 instead. Although, I am not sure how to do this in automake yet.

Reply:
Yes, I figure it could be done with a `--enable-static-nghttp2` option in a very similar way.

Reply:
Then should I do that instead?

Reply:
Well, I'm not convinced it is a terribly motivated change. You could also just invoke `CPPFLAGS=-DNGHTTP2_STATICLIB ./configure` and then you wound't have to change configure at all...

Reply:
Whatever you do, please don't tie it to BUILD_SHARED_LIBS. The fact that you want to build static curl does not imply that you want all the dependencies to be static.